File: 100 day challenge/Day2/Single Number II/SingleNumber.py
from typing import List
def main():
    v = [0,1,0,1,0,1,99]
    a=Solution();
    print(a.singleNumber(v))

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        d={x:nums.count(x) for x in nums}
        return list(d.keys())[list(d.values()).index(1)]

if __name__ == '__main__':
    main()

File: DS_Training/Data_structure/Array/SingleNumber.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DS_Training.Data_structure
{
    public static class SingleNumber
    {
        public static int solution(int[] nums)
        {
            HashSet<int> counter = new HashSet<int>(nums);
            return counter.Sum() * 2 - new List<int>(nums).Sum();
        }
    }
}
